core/src/main/java/com/datastax/oss/driver/api/core/session/SessionBuilder.java

```diff
@@ -671,8 +675,11 @@ public SessionT build() {
           defaultConfig.getStringList(DefaultDriverOption.CONTACT_POINTS, Collections.emptyList());
       if (cloudConfigInputStream != null) {
         if (!programmaticContactPoints.isEmpty() || !configContactPoints.isEmpty()) {
-          throw new IllegalStateException(
-              "Can't use withCloudSecureConnectBundle and addContactPoint(s). They are mutually exclusive.");
+          LOG.info(
+              "The withCloudSecureConnectBundle and addContactPoint(s) were provided. They are mutually exclusive. The addContactPoint(s) setting will be ignored.");
+          // clear the contact points provided in the setting file and via addContactPoints
+          configContactPoints = Collections.emptyList();
+          programmaticContactPoints = new HashSet<>();
         }
         String configuredSSLFactory =
```

Comment:
I am not sure if we should ignore (not fail) for SSL as well.
The `defaultConfig.getString(DefaultDriverOption.SSL_ENGINE_FACTORY_CLASS, null);` 
setting is used in multiple places, for example:
`DefaultDriverContext`:
```
protected Optional<SslEngineFactory> buildSslEngineFactory(SslEngineFactory factoryFromBuilder) {
    return (factoryFromBuilder != null)
        ? Optional.of(factoryFromBuilder)
        : Reflection.buildFromConfig(
            this,
            DefaultDriverOption.SSL_ENGINE_FACTORY_CLASS,
            SslEngineFactory.class,
            "com.datastax.oss.driver.internal.core.ssl");
  }
```
Ignoring this property will work now, but it seems fragile and may lead to some problems in the future (If we would change one of the usages of `DefaultDriverOption.SSL_ENGINE_FACTORY_CLASS`)
For `contactPoints` ignoring is not problematic because usage of `configContactPoints` and `programmaticContactPoints` is encapsulated in the `SessionBuilder`, wdty?

Reply:
Any SSL configuration will be overridden by line 694 if a secure connect bundle is present:

```
withSslEngineFactory(cloudConfig.getSslEngineFactory());
```

So I think we can just replace lines 687-688 by a warning and be done with it :-)

Reply:
However we should add a warning as well for local datacenter. Same pattern as for contact points.

Reply:
It is a bit problematic for contact points because it fails with:
```
 Multiple entries with same key: default=dc1 and default=dc-ignored
```
It is using the 
`
private ImmutableMap.Builder<String, String> localDatacentersBuilder = ImmutableMap.builder();
`
underneath and it does not support removing. Are we ok with changing this?

Reply:
You can add a new method `ProgrammaticArguments.Builder.clearDatacenters()`.